### DISCUSSION TITLE
Really executable JAR with spring-boot-maven-plugin

### DISCUSSION THREAD

**Discussion by tisonkun** [remkop/picocli#1903] [Show and tell]
https://github.com/korandoru/zeronos/blob/main/distribution/pom.xml

```xml
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
                <version>2.7.0</version>
                <executions>
                    <execution>
                        <id>repackage</id>
                        <goals>
                            <goal>repackage</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <mainClass>io.korandoru.zeronos.command.ZeronosMain</mainClass>
                    <executable>true</executable>
                </configuration>
            </plugin>
```

POM's packaging should be `jar` and I also do a post assembly work to rename the JAR file to an executable file, which is not required.

To try it out:

```bash
gh repo clone korandoru/zeronos
mvn clean install -DskipTests
./build-target/zeronos --help
```

This solution is not listed on the doc and the plugin is latest and maintained, while it still requires a installed JRE.

**Comment by remkop** [comment]
Would you like to contribute a pull request for the user manual `docs/index.adoc`?

**Reply by tisonkun** [reply]
@remkop I make a more impressive example at http://github.com/korandoru/hawkeye where I package picocli interface into executable JARs and GraalVM native image.

I'm going to write a paragraph about how to package with spring-boot-maven-plugin in days, but I also wonder if I can add this real-world use case somewhere. HawkEye is used for several projects and picocli acts as an important role to make it work.

**Reply by tisonkun** [reply]
FYI - https://github.com/remkop/picocli/pull/1991
